app/src/main/kotlin/de/janhopp/luebeckmensawidget/ui/MensaDayView.kt

```diff
@@ -1,22 +1,43 @@
 package de.janhopp.luebeckmensawidget.ui
 
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.glance.GlanceModifier
+import androidx.glance.LocalContext
 import androidx.glance.appwidget.lazy.LazyColumn
 import androidx.glance.appwidget.lazy.items
 import androidx.glance.layout.fillMaxSize
 import de.janhopp.luebeckmensawidget.api.model.MensaDay
-import de.janhopp.luebeckmensawidget.api.model.MensaLocation
+import de.janhopp.luebeckmensawidget.api.model.PriceGroup
+import de.janhopp.luebeckmensawidget.storage.Option
+import de.janhopp.luebeckmensawidget.storage.OptionsStorage
 
 @Composable
 fun MensaDayView(day: MensaDay) {
+    val options = OptionsStorage(LocalContext.current)
+    var showDate by remember { mutableStateOf(Option.ShowDate.defaultValue) }
+    var useEmoji by remember { mutableStateOf(Option.UseEmoji.defaultValue) }
+    var priceGroup by remember { mutableStateOf(Option.PriceGroup.defaultValue) }
+
+    LaunchedEffect(Unit) {
+        showDate = options.getBoolean(Option.ShowDate)
+        useEmoji = options.getBoolean(Option.UseEmoji)
+        priceGroup = options.getString(Option.PriceGroup)
+    }
```

Comment:
Because the loading needs a second, there is a flash of the layout with the default settings. Maybe this could be hidden by a loading indicator - maybe refactor a bit so that the loading indicator when updating also covers the loading of the settings?

Reply:
I tried to fix this by moving things around a bit. If it does not work, we will make it an issue later.